core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -364,6 +364,21 @@ public boolean hasMandatoryDependencies() {
         return dependencies.stream().anyMatch(d -> !d.optional);
     }
 
+    /**
+     * Is this plugin deprecated?
+     *
+     * @return {@code true} if and only if an update site reports deprecations for this plugin.
+     * @since TODO
+     */
+    @Restricted(NoExternalUse.class)
```

Comment:
Why is it restricted for external use?
I can see some usage in plugins that might want to know this information. Support Core for example.

Reply:
This is transient information. Whoever/whatever consumes Support Core data should just query update sites directly to get up-to-date information.